spancontext.go

```diff
@@ -218,6 +218,10 @@ func (c *SpanContext) SetHTTPRequest(req *http.Request) {
 		Name:     destinationServiceURL.String(),
 		Resource: destinationServiceResource,
 	})
+	c.SetServiceTarget(ServiceTargetSpanContext{
+		Type: "http",
```

Comment:
```suggestion
```

Now that your fallback PR has merged, and we set the type in `Span.End()`, I think we can omit Type here and it will be inferred as expected for Elasticsearch spans.

Reply:
I think that will only work for exit spans which is not the case in the tests

Reply:
Ah, so that highlights another issue actually: non-exit spans are not supposed to have `service.target.*` fields. The spec says, at https://github.com/elastic/apm/blob/main/specs/agents/tracing-spans-service-target.md#field-values:

> span.context.service.target.* fields should be omitted for non-exit spans.

So I think we should:
 - update the test to mark the span as an exit span
 - omit Type here
 - update the code to ignore the `c.SetServiceTarget` call for non-exit spans (perhaps in another PR)

WDYT?

Reply:
Good catch! Agree, I've updated the test for this PR